### PR TITLE
MGMT-10171: Form fields get invalidated immediately before entering information

### PR DESCRIPTION
**Pull request by batzionb**
https://issues.redhat.com/browse/MGMT-10171

Right now we show errors on an input field if it was touched.
This causes required array fields new items to become invalid right away.
Instead I propose the following:
If the value is empty - show the error only if it was previously none empty
If the value is not empty - show the error 
This change effects all input fields across the lib

### CODE REVIEW

File: src/common/hooks/useFieldErrorMsg.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  validate?: FieldValidator;
+};
+
+const useFieldErrorMsg = ({ name, inputError, validate }: Props): string | undefined => {
+  /*
+    for empty values - show an error only if at a certain point it was none empty, to avoid new items in a field array shown as invalid right away
+    for none empty values - always show the error if there is one
+  */
+  const [field, { error }] = useField({
+    name: name,
```
Comment (jkilzi) [inline src/common/hooks/useFieldErrorMsg.tsx:16]:
```suggestion
    name,
```

File: src/common/hooks/useFieldErrorMsg.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { FieldValidator, useField } from 'formik';
+
+type Props = {
+  name: string;
+  inputError?: string;
+  validate?: FieldValidator;
+};
+
+const useFieldErrorMsg = ({ name, inputError, validate }: Props): string | undefined => {
```
Comment (jkilzi) [inline src/common/hooks/useFieldErrorMsg.tsx:10]: Am I missing something? Seems like `validate` is not in use here
```suggestion
const useFieldErrorMsg = ({ name, inputError }: Props): string | undefined => {
```

File: src/common/hooks/useFieldErrorMsg.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    validate,
+  });
+  const [hadValue, setHadValue] = React.useState<boolean>(false);
+  React.useEffect(() => {
+    if (field.value) {
+      setHadValue(field.value);
+    }
+  }, [field.value, setHadValue]);
+  const showError = field.value || (!field.value && hadValue);
+  const errorMessage = (showError ? error : '') || inputError;
```
Comment (jkilzi) [inline src/common/hooks/useFieldErrorMsg.tsx:26]: Can `error` be undefined? The same for `inputError`

**Review by jkilzi — CHANGES_REQUESTED** [review]

File: src/common/hooks/useFieldErrorMsg.tsx (hunk 2 above)
Comment (batzionb) [reply src/common/hooks/useFieldErrorMsg.tsx:10]: validate is passed to useField

File: src/common/hooks/useFieldErrorMsg.tsx (hunk 3 above)
Comment (batzionb) [reply src/common/hooks/useFieldErrorMsg.tsx:26]: yes they can both be undefined

File: src/common/hooks/useFieldErrorMsg.tsx (hunk 2 above)
Comment (jkilzi) [reply src/common/hooks/useFieldErrorMsg.tsx:10]: oops... i overlooked that :) 10x

File: src/common/hooks/useFieldErrorMsg.tsx (hunk 3 above)
Comment (jkilzi) [reply src/common/hooks/useFieldErrorMsg.tsx:26]: ok, so I guess is up to the caller to protect against this case

**Review by jkilzi — APPROVED** [review]

**Pull request merged**